corehq/apps/userreports/expressions/specs.py

```diff
@@ -304,3 +305,29 @@ def _get_forms(self, case_id, context):
 
         context.set_cache_value(cache_key, xforms)
         return xforms
+
+
+class SubcasesExpressionSpec(JsonObject):
+    type = TypeProperty('get_subcases')
+    case_id_expression = DefaultProperty(required=True)
+
+    def configure(self, case_id_expression):
+        self._case_id_expression = case_id_expression
+
+    def __call__(self, item, context=None):
+        case_id = self._case_id_expression(item, context)
+        if not case_id:
+            return []
+
+        assert context.root_doc['domain']
+        return self._get_subcases(case_id, context)
+
+    def _get_subcases(self, case_id, context):
+        domain = context.root_doc['domain']
+        cache_key = (self.__class__.__name__, case_id)
+        if context.get_cache_value(cache_key) is not None:
+            return context.get_cache_value(cache_key)
```

Comment:
I know fetching from cache is supposed to be cheap but it's still a network round trip + IO. Would it not be better to just assign the value to a variable rather than fetching it twice?